tests/test_feature.py

```diff
@@ -910,15 +946,38 @@ def test_signal_sliding_window(win_dur, hop_dur):
     ends = starts + win_dur
 
     index = audinterface.utils.signal_index(starts, ends)
-    pd.testing.assert_frame_equal(
-        features,
-        pd.DataFrame(
-            expected_features,
-            index=index,
-            columns=extractor.column_names,
-        ),
+    expected = pd.DataFrame(
+        np.ones((n_time_steps, len(interface.column_names))),
+        index=index,
+        columns=interface.column_names,
+    )
+    pd.testing.assert_frame_equal(df, expected)
+
+
+def test_signal_sliding_window_error():
+
+    interface = audinterface.Feature(
+        feature_names='mean',
+        process_func=mean_sliding_window,
+        process_func_args={
+            'win_dur': 0.5,
+            'hop_dur': 0.25,
+        },
+        process_func_is_mono=False,
+        process_func_applies_sliding_window=True,
```

Comment:
Why do we need the `process_func_applies_sliding_window` argument?
We have set the window and hop duration anyway as parameter of the sliding window. Or is it just to add a check to ignore the `win_dur` and `hop_dur` arguments inside `Feature`?

Reply:
We need some way to tell `Feature` whether the process function will apply the sliding window or if this should be handled internally. So do you propose we should derive that from the arguments to the processing function?

Reply:
The other problem is that even if the processing function does not apply the sliding window itself, it may still need the information about the window and hop duration. In that case it would be a bad idea to automatically derive the strategy from the fact that the processing function has a `win_dur` and `hop_dur` argument. So guess it's safer to introduce an argument to control the behavior.

However, I now implemented that the `win_dur` and `hop_dur` arguments are automatically added to `process_func_args` if the processing function expects such arguments and they are not explicitly set by the user. Now:

```python
db = audb.load(
    'emodb',
    version='1.2.0',
    format='wav',
    mixdown=True,
    sampling_rate=16000,
    full_path=False,
)


def mean(signal, sampling_rate, win_dur, hop_dur):
    frames = audinterface.utils.sliding_window(
        signal,
        sampling_rate,
        win_dur,
        hop_dur,
    )
    return frames.mean(axis=1, keepdims=True)


win_dur=1.0
hop_dur=0.5

interface = audinterface.Feature(
    'mean',
    process_func=mean,
    process_func_applies_sliding_window=True,
    process_func_args={
        'win_dur': win_dur,
        'hop_dur': hop_dur,
    },
    win_dur=win_dur,
    hop_dur=hop_dur,
)
interface.process_files(db.files[:3], root=db.root)
```
```
                                                                   mean
file            start                  end                             
wav/03a01Fa.wav 0 days 00:00:00        0 days 00:00:01        -0.000329
                0 days 00:00:00.500000 0 days 00:00:01.500000 -0.000285
wav/03a01Nc.wav 0 days 00:00:00        0 days 00:00:01         0.000039
                0 days 00:00:00.500000 0 days 00:00:01.500000 -0.000412
wav/03a01Wa.wav 0 days 00:00:00        0 days 00:00:01        -0.000284
                0 days 00:00:00.500000 0 days 00:00:01.500000 -0.000318
```

can be simplified to:

```python
...
interface = audinterface.Feature(
    'mean',
    process_func=mean,
    process_func_applies_sliding_window=True,
    win_dur=win_dur,
    hop_dur=hop_dur,
)
...
```

At least there is a way to avoid setting `win_dur` and `hop_dur` twice.

I would then recommend that we immediately deprecate the default value of `process_func_applies_sliding_window` from `True` to `False`.

Reply:
> What's also unfortunate is that - again for backward compatibility - we have to assume by default that the processing function will apply the sliding window, so the default behavior is different from what we have in Process.

In `Process` we don't have the `process_func_applies_sliding_window`, so I guess we don't need to have this argument there and the process func can apply a sliding window or not if we set `win_dur=None`?

Reply:
In `Process` such an argument would not make sense, since we do assume that the returned values have a certain shape, i.e. we cannot know which axis should be expand into frames.